pkg/blockstorage/azure/client.go

```diff
@@ -49,7 +51,20 @@ var (
 	subscriptionsClientFactory *armsubscriptions.ClientFactory
 )
 
-func NewClient(ctx context.Context, config map[string]string) (*Client, error) {
+type ComputeNewClientFactoryFunc func(subscriptionID string, credential azcore.TokenCredential, options *arm.ClientOptions) (*armcompute.ClientFactory, error)
+type SubscriptionsNewClientFactoryFunc func(credential azcore.TokenCredential, options *arm.ClientOptions) (*armsubscriptions.ClientFactory, error)
+type AzClientFactoriesType struct {
+	ComputeNewClientFactory       ComputeNewClientFactoryFunc
+	SubscriptionsNewClientFactory SubscriptionsNewClientFactoryFunc
+}
+
+func NewClient(ctx context.Context, config map[string]string, azClientFactories *AzClientFactoriesType) (*Client, error) {
```

Comment:
This function is only called as `nil` as a third argument

Reply:
It is overridden in unit tests